File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(vector<string> strArr) {
    vector<string> answer;
    for(int i=0; i<strArr.size(); i++) {
        bool flag = false;
        for(int j=0; j<strArr[i].size(); j++) {
            if (strArr[i].substr(j, 2) == "ad") flag = true;
        }
        if (!flag) answer.push_back(strArr[i]);
    }
    return answer;
}

File: 프로그래머스/0/181870. ad 제거하기/ad 제거하기.java
import java.util.*;
class Solution {
    public String[] solution(String[] strArr) {
        List<String> li = new ArrayList<>();
        for(int i=0; i<strArr.length; i++) {
            if(strArr[i].contains("ad")) continue;
            li.add(strArr[i]);
        }
        String[] answer = new String[li.size()];
        for(int i=0; i<li.size(); i++) {
            answer[i] = li.get(i);
        }
        return answer;
    }
}